main.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+)
+
+const (
+	PORT = "8080"
+)
+
+func logger(fun http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		start := time.Now()
+		log.Printf("%s %s %s", r.RemoteAddr, r.Method, r.URL)
+		fun(w, r)
+		log.Printf("%s %s %s duration %s", r.RemoteAddr, r.Method, r.URL, time.Since(start))
+	}
+}
+
+func corsMiddleware(next http.Handler) http.Handler {
+	allowedOrigins := map[string]bool{
+		"http://localhost:3000": true,
+		"http://127.0.0.1:3000": true,
+		// production domains...
+	}
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		origin := r.Header.Get("Origin")
+		if origin != "" && allowedOrigins[origin] {
+			w.Header().Set("Access-Control-Allow-Origin", origin)
+			w.Header().Set("Vary", "Origin")
+			w.Header().Set("Access-Control-Allow-Credentials", "true")
+			w.Header().Set("Access-Control-Allow-Methods", "GET, POST, PUT, DELETE, OPTIONS")
+			w.Header().Set("Access-Control-Allow-Headers", "Content-Type, Authorization, Accept, Origin, X-Requested-With, X-CSRF-Token")
+		} else {
+			// origin отсутствует (обычный запрос) или не разрешён - не выставляем заголовок
+		}
```

Comment:
Вот тут немного не так.

Origin заголовок - это откуда у тебя был выслан запрос. 
Access-Control-Allow-Origin - заголовок разрешающий сложные запросы с определенных источников. 

 Ты должен просто заголовки выставить всегда фиксированный origin, например "http://localhost:3000,http://127.0.0.1:3000" а браузер сам проверит. 
 
```suggestion
			w.Header().Set("Access-Control-Allow-Origin", origin)
			w.Header().Set("Vary", "Origin")
			w.Header().Set("Access-Control-Allow-Credentials", "true")
			w.Header().Set("Access-Control-Allow-Methods", "GET, POST, PUT, DELETE, OPTIONS")
			w.Header().Set("Access-Control-Allow-Headers", "Content-Type, Authorization, Accept, Origin, X-Requested-With, X-CSRF-Token")
```

Reply:
Т.е. это защита от выполнения запросов к твоему серверу со сторонних != Access-Control-Allow-Origin сайтов в браузере. За пределами браузера запрос кто угодно может сделать в твой эндпоинт

Reply:
Сделано на небольшом костыле по последним твоим комментариям в ТГ